composer/core/surgery.py

```diff
@@ -299,3 +299,47 @@ def update_params_in_optimizer(old_params: Iterable[torch.nn.parameter.Parameter
     new_param_list += list(added_params)
     log.info(f'adding {len(added_params)} new parameters to parameter group #{group_idx}')
     param_group['params'] = new_param_list
+
+
+def replace_params_in_optimizer(old_params: Iterable[torch.nn.parameter.Parameter],
+                                new_params: Iterable[torch.nn.parameter.Parameter], optimizers: Optimizers) -> None:
+    """Fully replaces parameters from an optimizer.
```

Comment:
Fair point - I could move this into the trainer-specific code. I like keeping it here just because it's semantically related to surgery, and it seems likely we may refactor surgery some day so that `update_params_in_optimizer` and `replace_params_in_optimizer` can be unified someday. And who knows, maybe someone will write an algorithm some day that actually requires `replace_params_in_optimizer`.